lib/ps.js

```diff
@@ -1,131 +1,103 @@
+  */
+function ps (pids, options, done) {
+  var pArg = pids.join(',')
 
-  ps.stdout.on('data', function (d) {
-    stdout += d.toString()
-  })
+  var args = ['-o', 'etime,pid,ppid,pcpu,rss,time', '-p', pArg]
 
-  ps.stderr.on('data', function (d) {
-    stderr += d.toString()
-  })
+  if (PLATFORM === 'aix') {
+    args = ['-o', 'etime,pid,ppid,pcpu,rsssize,time', '-p', pArg]
+  }
 
-  ps.on('close', function (code) {
+  bin('ps', args, function (err, stdout) {
+    if (err) return done(err)
     var date = Date.now()
-    if (error) return done(error, null)
-    if (code !== 0) return done(new Error('ps exited with code ' + code + ', command was "ps ' + args.join(' ') + '" and error: ' + stderr))
-    if (!stdout) {
-      return done(null, pids.map(function (pid) {
-        return {cpu: 0, memory: 0, pid: parseInt(pid, 10), time: 0, start: new Date(date)}
-      }))
-    }
+
+    // Example of stdout on *nix.
+    // ELAPSED: format is [[dd-]hh:]mm:ss
+    // RSS: is counted as blocks of 1024 bytes
+    // TIME: format is [[dd-]hh:]mm:ss
+    //
+    //    ELAPSED   PID  PPID  %CPU     RSS        TIME
+    // 2-40:50:53   430     1   3.0    5145  1-02:03:04
+    //   40:50:53   432   430   0.0    2364  1-01:02:03
+    //   01:50:50   727     1  10.0  348932       14:27
+    //      00:20  7166     1   0.1    3756        0:00
+
+    // Example of stdout on Darwin
+    // ELAPSED: format is [[dd-]hh:]mm:ss
+    // RSS: is counted as blocks of 1024 bytes
+    // TIME: format is [[dd-]hh:]ss:mm.pp (pp is the percentage of a minute)
+    //
+    //    ELAPSED   PID  PPID  %CPU     RSS           TIME
+    // 2-40:50:53   430     1   3.0    5145  1-02:03:04.07
+    //   40:50:53   432   430   0.0    2364  1-01:02:03.10
+    //   01:50:50   727     1  10.0  348932       14:27.26
+    //      00:20  7166     1   0.1    3756        0:00.02
 
     stdout = stdout.split(os.EOL)
 
     var statistics = {}
     for (var i = 1; i < stdout.length; i++) {
-      stdout[i] = stdout[i].trim()
+      var line = stdout[i].trim().replace(/\s\s+/g, ' ').split(' ')
 
-      if (!stdout[i]) {
+      if (!line || line.length !== 6) {
         continue
       }
 
-      var line = stdout[i].replace(/^\s+/, '').replace(/\s\s+/g, ' ').split(' ')
-      var id = parseInt(line[2], 0)
-      var time = doParseTime ? parseTime(line[3]) : parseInt(line[3], 10)
-
-      statistics[id] = {
-        cpu: parseFloat(line[0].replace(',', '.')),
-        memory: parseFloat(line[1]) * 1024,
-        pid: id,
-        time: time, // time in seconds
-        start: new Date(date - time * 1000)
+      var pid = parseInt(line[1], 10)
+      var ppid = parseInt(line[2], 10)
+      var cpu = parseFloat(line[3].replace(',', '.'), 10)
+      var memory = parseInt(line[4], 10) * 1024
+      var etime = parseTime(line[0])
+      var ctime = parseTime(line[5], true)
+
+      statistics[pid] = {
+        cpu: Math.min(Math.round(cpu * 1000) / 1000, 100.0),
```

Comment:
you said in the readme that it could be over 100? why don't just give back the results?
Also what's the point of the Math.round?

Reply:
Because we know that it cannot be greater than 100.

The point of Math.round is to round to 2 decimal places.
The percentage is just an approximation there's no need to have more than 2 digits of precision.